src/ixdat/plotters/ms_plotter.py

```diff
@@ -266,35 +356,77 @@ def plot_vs(
                     tspan_bg=tspan_bg,
                     remove_background=remove_background,
                     include_endpoints=False,
+                    return_quantity=use_quantity,
+                )
+
+            if ureg("mol / s / cm ** 2").check(unit):
+                if not hasattr(measurement, "A_el"):
+                    warnings.warn(
+                        "Measurement does not have an attribute A_el and cannot "
+                        "calibrate to A_el"
+                    )
+                    continue
+                if isinstance(measurement.A_el, type(Quantity)):
+                    v = v / measurement.A_el
+                else:
+                    v = v / (measurement.A_el * ureg.cm**2)
+                    warnings.warn(
+                        f"You should explicit set the unit of A_el: {measurement.A_el} "
+                        f"before plotting. Falling back to default unit {ureg.cm**2:~P}"
+                    )
+
+            if use_quantity and not v.check(unit) and not logdata:
+                vn = f"n_dot_{v_name}" if quantified else f"{v_name}"
+                error_message = (
+                    f"cannot plot {vn} with units {v.u:~P} on axis with units {unit}."
+                    f"If the unit of '{vn}' is wrong you can set the correct unit with "
+                    f"measurement['{vn}'].unit.set_unit('correct_unit_name')"
+                    "Otherwise choose a compatible unit to plot on axis. List of "
+                    f"compatible units to dataseries: {ureg.get_compatible_units(v.u)}"
                 )
+                print(error_message)
+                continue
+
             if logplot:
-                v[v < MIN_SIGNAL] = MIN_SIGNAL
+                try:
+                    v.m[v.m < MIN_SIGNAL] = MIN_SIGNAL
+                except (DimensionalityError, AttributeError):
+                    v[v < MIN_SIGNAL] = MIN_SIGNAL
+
+            if logdata:
+                logplot = False
+                vu = v.to(unit).u if use_quantity else unit
+                v = (
+                    np.log(v.to(unit).m) * ureg.dimensionless
+                    if use_quantity
+                    else np.log(v / (1 * unit).to_base_units().m) * ureg.dimensionless
+                )
+                ylabel = f"{vu:~ixdat_log}"
+
             x_mass = np.interp(t_v, t, x)
```

Comment:
I know this isn't you but `x_mass` is a wierd name... looks like it should be called `x_v` (x for the specific value being plotted) to be consistent with the nomenclature `t_v`.